src/main/java/org/broadinstitute/hellbender/tools/spark/pipelines/FlagStatSpark.java

```diff
@@ -0,0 +1,78 @@
+import org.broadinstitute.hellbender.cmdline.argumentcollections.OptionalIntervalArgumentCollection;
+import org.broadinstitute.hellbender.cmdline.argumentcollections.ReadInputArgumentCollection;
+import org.broadinstitute.hellbender.cmdline.argumentcollections.RequiredReadInputArgumentCollection;
+import org.broadinstitute.hellbender.cmdline.programgroups.DataFlowProgramGroup;
+import org.broadinstitute.hellbender.cmdline.programgroups.SparkProgramGroup;
+import org.broadinstitute.hellbender.engine.spark.SparkCommandLineProgram;
+import org.broadinstitute.hellbender.engine.spark.datasources.ReadsSparkSource;
+import org.broadinstitute.hellbender.exceptions.UserException;
+import org.broadinstitute.hellbender.tools.FlagStat;
+import org.broadinstitute.hellbender.tools.FlagStat.FlagStatus;
+import org.broadinstitute.hellbender.utils.IntervalUtils;
+import org.broadinstitute.hellbender.utils.SimpleInterval;
+import org.broadinstitute.hellbender.utils.dataflow.BucketUtils;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.io.PrintStream;
+import java.util.List;
+
+@CommandLineProgramProperties(summary ="runs FlagStat on Spark", oneLineSummary = "FlagStat", programGroup = SparkProgramGroup.class)
+public final class FlagStatSpark extends SparkCommandLineProgram {
+
+    private static final long serialVersionUID = 1L;
+
+    @ArgumentCollection
+    public ReadInputArgumentCollection readArguments= new RequiredReadInputArgumentCollection();;
+
+    @ArgumentCollection
+    public IntervalArgumentCollection intervalArgumentCollection = new OptionalIntervalArgumentCollection();
+
+    @Argument(doc = "uri for the output file: a local file path",
+            shortName = StandardArgumentDefinitions.OUTPUT_SHORT_NAME, fullName = StandardArgumentDefinitions.OUTPUT_LONG_NAME,
+            optional = true)
+    public String out;
+
+    @Override
+    protected void runPipeline(final JavaSparkContext ctx) {
+        if ( readArguments.getReadFilesNames().size() != 1 ) {
+            throw new UserException("This tool only accepts a single bam/sam/cram as input");
+        }
+        final String bam = readArguments.getReadFilesNames().get(0);
+        final ReadsSparkSource readSource = new ReadsSparkSource(ctx);
+        SAMFileHeader readsHeader = ReadsSparkSource.getHeader(ctx, bam);
+
+        /**
+         * If no intervals specified - use all reads, mapped and unmapped.
+         */
+        final List<SimpleInterval> intervals = intervalArgumentCollection.intervalsSpecified() ? intervalArgumentCollection.getIntervals(readsHeader.getSequenceDictionary())
+                : null;
+
+        final JavaRDD<GATKRead> reads = readSource.getParallelReads(bam, null);
+        final FlagStatus result = reads.map(r -> new FlagStatus().add(r)).fold(new FlagStatus(), (fs1, fs2) -> fs1.merge(fs2));
```

Comment:
it might be faster to do 
```
final FlagStatus result = reads.aggregate(new FlagStatus(), FlagStatus::add, FlagStatus::merge);
```
if I understand correctly this will only create 1 `FlagStatus` per partition, merge all the reads into it, and then merge the `FlagStatus`s at the end. 

@laserson What's the preferred method for these things?